datafusion/physical-plan/src/joins/nested_loop_join.rs

```diff
@@ -718,8 +718,6 @@ struct NestedLoopJoinStream<T> {
     inner_table: OnceFut<JoinLeftData>,
     /// Information of index and left / right placement of columns
     column_indices: Vec<ColumnIndex>,
-    // TODO: support null aware equal
```

Comment:
Rather than just deleting this, Maybe we can add a comment explaining the rationale for why it doesn't need null_equals_null.

```suggestion
    // No need to support null aware equal because ...
```